app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesActivity.kt

```diff
@@ -148,19 +148,19 @@ class SearchRepositoriesActivity : AppCompatActivity() {
 
```

Comment:
The reason is due to a race condition where invalidation is triggered after remote refresh returns - this happens in Room where invalidation callbacks require a transaction so it can get blocked. If source.refresh is already idle, it could trigger the helper to transition to NotLoading early. This cannot be fixed in the library because we cannot assume remote refresh will trigger invalidation even if it returns success... for example if it fetches items from network and there are no new items. Room will always invalidate on any write, but this isn't an assumption we can make in general...

Also since you have distinctUntilChanged above, you can just use filter instead of .map so you don't trigger .combine as often later

So you should use:

```
        val notLoading = repoAdapter.loadStateFlow
            // Only emit when REFRESH LoadState for RemoteMediator changes.
            .distinctUntilChangedBy { it.mediator.refresh }
            // Only react to cases where Remote REFRESH completes i.e., NotLoading.
            .filter { it.mediator.refresh is LoadState.NotLoading }
```